FoxHound.Client/src/Posts/Post.tsx

```diff
@@ -41,6 +41,12 @@ const Post: React.FC = () => {
     }
   };
 
+  const commentAdded = (comment: CommentModel) => {
+    const postCopy = { ...post };
+    postCopy.comments.push(comment);
+    setPost(postCopy);
+  };
```

Comment:
I'm curious on the approach here. Currently, this is adding the comment to array of comments that live on the post. Should this page maintain 2 different states, one for the post, and one for the comment array? I'm curious if there is a best practice here. I think the way this is currently will force the entire post to re-render when a comment is added. What happens if on a given comment, we want to change the state (i.e. like a comment), how does this effect the pattern?

Reply:
So first off, because you are referencing previous state, you should put all that code in a callback function as an argument to setPost(prevState => { ... }).

Reply:
But to answer your question, I don't know that I would be too worried about re-rendering the post each time. I can't really think of any reasonable way around it anyways. Be interested what @chrispollaro thought.

Reply:
I don't think it would cause a big difference like Ryan said; however, I think React may be smart enough to only re-render elements in the shadow DOM that it actually needs to, even if a React component is affected, I think it will only touch the necessary elements.  Say you had a component with two input fields with their own state.  I don't think it rerender the top field if the state only affected the bottom one.

Reply:
I'm updated the setPost(...) to use the callback with the prevState. I'm still a little confused on the approach here in general, but i'm hoping it will work itself out as we keep adding features to this. Would @ryan-cat or @chrispollaro  approve this PR if you don't see any other issues?